### ISSUE TITLE
Deep Patient: An Unsupervised  Representation to Predict the  Future of Patients from the  Electronic Health Records

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#181]
## Abstract
_**"a three-layer stack of denoising autoencoders to capture hierarchical regularities and dependencies in the aggregated EHRs"**_

## Introduction
* EHR data
  * high dimensionality, noise, heterogeneity, sparseness, incompleteness, random errors, systematic biases
  * the same clinical phenotype can be expressed using different codes and terminologies
* predictive algorithms largely depends on feature selection and data representation
* “deep patient”
  * represent patients by a set of general features, which are inferred automatically from a large-scale EHR database through a deep learning approach

### Deep Patient Representation
* 1. EHRs are extracted from the clinical data warehouse
* 2. pre-processed
* 3. grouped in patient vectors (=> raw representation)
* 4. used as input of the feature learning algorithm
* 5. represented using features
  * deep representation can be applied to different clinical tasks
  * derived the patient representation using a multi-layer neural network in a deep learning architecture
* The last network of the chain outputs the final patient representation

### Denoising Autoencoders
* a stack of denoising autoencoders (SDA) 사용.
  * independently trained layer by layer
  * share the same structure and functionalities
* 인코더
  <image src="https://user-images.githubusercontent.com/67627471/135754704-6be4e4d8-c67b-4c76-820d-1051b84a3303.png" width="75%" height="75%" />
* 디코더
  <image src="https://user-images.githubusercontent.com/67627471/135754724-462ada24-d31a-4fa9-aa9f-079ce98ec7e3.png" width="60%" height="60%" />
* 인코딩 함수 결과인 y와 디코딩 함수 결과인 z 사이의 difference를 최소화하는 parameters 찾도록 함.
  <image src="https://user-images.githubusercontent.com/67627471/135754831-b20d1ef6-03f7-435d-a0ec-0d42f15b09ac.png" width="60%" height="60%" />
* 오버피팅 방지로 Autoencoders는 noisy 버전에서 input 형태로 recon하는 것도 학습함.
  <image src="https://user-images.githubusercontent.com/67627471/135754850-b1fb559b-4844-445f-bdfd-acbb50e3eff8.png" width="60%" height="60%" />

### 데이터셋
* retained all patients with at least one diagnosed disease expressed as numerical ICD-9 between 1980 and 2014
* --> a dataset of about 1.2 million patients, with every patient having an average of 88.9 records
* considered all records up to December 31, 2013 (=> “split-point”로 정함.) as training data
* all the diagnoses in 2014 as testing data

### Future Disease Prediction
* implemented random forest classifiers trained over each disease using a dataset of 200,000 patients
* computed the probability to develop every disease in the vocabulary
* => each patient was represented by a vector of disease probabilities

## Results
* Disease classification results in terms of area under the ROC curve (AUC-ROC), accuracy and F-score
  <image src="https://user-images.githubusercontent.com/67627471/135754975-12a338b7-1bc4-40cc-8a87-8730978919aa.png" width="50%" height="50%" />

##
* EHR data의 representation을 unsupervised learning으로 학습한다는 점에서 어제 읽은 논문과 유사.
* autoencoder과 random forest를 사용한 것도 동일. (적용 방법은 약간 다르지만)
